File: Kth Smallest Element in BST/Kth Smallest Element In BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.count = 0
        self.k = k
        self.inorder(root);
        return self.ans;
    
    def inorder(self, node):
        if (node.left is not None):
            self.inorder(node.left)
        
        self.count += 1
        if (self.count == self.k):
            self.ans = node.val;
            return;
        
        if (node.right is not None):
            self.inorder(node.right);

File: Kth Smallest Element in BST/Kth Smallest Element In BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int count = 0;
    public int ans = -1;
    public int kthSmallest(TreeNode root, int k) {
        inOrder(root, k);
        return ans;
    }
    
    public void inOrder(TreeNode node, int k) {
        if (node.left != null) {
            inOrder(node.left, k);
        }
        count++;
        if (count == k) {
            ans = node.val;
            return;
        }
        if (node.right != null) {
            inOrder(node.right, k);
        }
        
    }
}